homeassistant/components/mqtt/__init__.py

```diff
@@ -670,22 +711,26 @@ def _match_topic(subscription, topic):
 class MqttAvailability(Entity):
     """Mixin used for platforms that report availability."""
 
-    def __init__(self, availability_topic, qos, payload_available,
-                 payload_not_available):
+    def __init__(self, availability_topic: Optional[str], qos: Optional[int],
+                 payload_available: Optional[str],
+                 payload_not_available: Optional[str]) -> None:
         """Initialize the availability mixin."""
         self._availability_topic = availability_topic
         self._availability_qos = qos
-        self._available = availability_topic is None
+        self._available = availability_topic is None  # type: bool
         self._payload_available = payload_available
         self._payload_not_available = payload_not_available
 
+    @asyncio.coroutine
     def async_added_to_hass(self):
```

Comment:
This clearly should be a coroutine afaik, don't know how/why this worked before...

Reply:
Each coroutine function returns a coroutine object. A coroutine object is a pending piece of work that will be executed when run on the event loop or yielded from a task that is being run on the event loop. So sometimes we used to not add `@asyncio.coroutine` and just return the object we get back from calling a function. It's however something we're going to stop doing once we're switching to Python 3.5.

Sometimes we also just forgot the decorator. However, asyncio once started as generators (the `yield` keyword) being overloaded in Python before it got standardized as `@asyncio.coroutine` and later `async`/`await`. So for backwards compatibility, one can still yield from generators. And this function because a generator because there was a `yield` statement inside the body.

Anyway, good you added the decorator 👍

Reply:
Oh thanks. I was already wondering what the `@asyncio.coroutine` decorator did, but `await` support makes perfect sense 👍